docs: change contribution guide about the product management (#5023)
zitadel/zitadel: CONTRIBUTING.md

## Product Management

The ZITADEL Team works with an agile product management methodology.
You can find all the issues prioritized and ordered in the [product board](https://github.com/orgs/zitadel/projects/2/views/1).

> Every two weeks the team goes through all the new issues and decided about the priority, effort and if it is ready to start or in the backlog.

Reviewer: ```suggestion
Every two weeks the team goes through new issues and decides on the priority, effort and if the issue is ready to start or goes in the backlog.
```

> Every two weeks the team goes through all the new issues and decided about the priority, effort and if it is ready to start or in the backlog.

Reviewer: Could we add when the sprint is starting (Tuesdays?) and how that impacts release (next monday?). Would just be good to have somewhere, so we can link or look it up.

Reviewer: Ah just found the part below (maybe mention: "see below")

If it is something critical or urgent we will have a look at it earlier.
To show the community the needed information, each issue gets attributes and labels.

## About the attributes

You can find the attributes on the project "Product Management".

### State

The state should reflect the progress of the issue and what is going on right now.

- **No status**: Issue just got added and has to be looked at.

> - **🧐 Investigating**: We are curently investigating to find out what the problem is, which priority it should have and what has to be implemented. Or we need some more information of the author.

Reviewer: ```suggestion
- **🧐 Investigating**: We are currently investigating to find out what the problem is, which priority it should have and what has to be implemented. Or we need some more information from the author.
```

> - **📨 Product Backlog**: If an issue is in the backlog, it is not currently being worked on. These are recorded so that they can be processed in the future. Issues with this state do not have to be completely defined yet.

Reviewer: ```suggestion
- **📨 Product Backlog**: If an issue is in the backlog, it is not currently being worked on. These are recorded so that they can be worked on in the future. Issues with this state do not have to be completely defined yet.
```

- **📝 Prioritized Product Backlog**: An issue with the state "Prioritized Backlog" is ready for the refinement from the perspective of the product owner (PO) to implement. This means the developer can find all the relevant information and acceptance criteria in the issue.
- **🔖 Ready**: The issue is ready to take into a sprint. Difference to "prioritized..." is that the complexity is defined by the team.
- **📋 Sprint Backlog**: The issue is scheduled for the current sprint.

> - **🏗 In progress**: Someone is working on this issue right now.

Reviewer: At which point does an issue get an assignee? Sprint Backlog or In Progress?

Author: In progress

> - **👀 In review**: The issue is in review.

Reviewer: When going from in progress to in review, do I have to request someone to review or is this done by PO? Would be maybe worth it to explicitly define the desired process.

Reviewer: Also how external contributors should do it

- **✅ Done**: The issue is implemented and merged to main.

### Priority

> Priority shows you the priority the ZITADEL team has given this issue. In general the more customers want a feature the higher the priority gets.

Reviewer: ```suggestion
Priority shows you the priority the ZITADEL team has given this issue. In general the higher the demand from customers and community for the feature, the higher the priority.
```

> - **🌋 Critical**: This is a security issue or something that has to be fixed urgently, because customers can't work anymore.

Reviewer: ```suggestion
- **🌋 Critical**: This is a security issue or something that has to be fixed urgently, because the software is not usable or highly vulnerable.
```

> - **🏔 High**: These are the issues the ZITADEL Team is currently focusing on and will be implemented as soon as possible.

Reviewer: ```suggestion
- **🏔 High**: These are the issues the ZITADEL team is currently focusing on and will be implemented as soon as possible.
```

- **🏕 Medium**: After all the high issues are done these will be next.
- **🏝 Low**: This is low in priority and will probably not be implemented in the next time or just if someone has some time in between.

### Complexity

This should give you an indication how complex the issue is. It's not about the hours or effort it takes.
Everything that is higher than 8 should be split in smaller parts.

**1**, **2**, **3**, **5**, **8**, **13**

### Sprint

> We want to deliver a new release every second week. So we plan everything in two week sprint.

Reviewer: As mentioned above: Maybe add start / end day and planned release periodicity of the sprints.

Author: I have changed the structure. Not sure why I placed this part here, makes more sense at the top

## About the Labels

There are a few general labels that don't belong to a specific category.

> - **good first issue**: This label shows contibuters, that it is an easy entry point to start developing on ZITADEL.

Reviewer: ```suggestion
- **good first issue**: This label shows contributors, that it is an easy entry point to start developing on ZITADEL.
```
